tez-mapreduce/src/main/java/org/apache/tez/mapreduce/committer/MROutputCommitter.java

```diff
@@ -78,6 +79,7 @@ public void initialize() throws IOException {
     jobConf.getCredentials().mergeAll(UserGroupInformation.getCurrentUser().getCredentials());
     jobConf.setInt(MRJobConfig.APPLICATION_ATTEMPT_ID,
         getContext().getDAGAttemptNumber());
+    jobConf.set(MRJobConfig.JOB_COMMITTER_UUID, Utils.getDAGID(getContext()));
```

Comment:
this unique across all jobs which may be writing to a table, even from other processes?

Reply:
Yes. This ID is unique to a DAG + attempt number - so if we have some other job, it'll have a different application ID component, while if an attempt fails and the DAG retries, the attempt number will be different.